### PR TITLE
Allow loadbalancer service external IP

### PR DESCRIPTION
**Pull request by YonatanRubin**

### CODE REVIEW

**Comment by mustafaStakater** [comment]
make a dummy commit so that latest pr pipeline is run `git commit -m "trigger pipeline" --allow-empty`

**Comment by rasheedamir** [comment]
@aslafy-z can you plz review it?

**Review by d3adb5 — CHANGES_REQUESTED** [review]
I suggested a few minor changes, one of which looks like a necessary fix. Let me know if you need help implementing any of this, @YonatanRubin. Thank you for working on this!

**Comment by rasheedamir** [comment]
@YonatanRubin can you plz take care of the comments?

**Comment by YonatanRubin** [comment]
Hi, sorry I will not be able to handle this in the next couple of weeks, I will work on this then. If you prefer you can close the PR and I will open a new one when I can.

**Comment by rasheedamir** [comment]
@YonatanRubin you can fix when you have time no problem

**Comment by rasheedamir** [comment]
@d3adb5 @aslafy-z can anyone of you please finalize this PR and get it merged?

File: application/templates/service.yaml [hunk 1]
```diff
@@ -7,20 +7,35 @@ metadata:
...
+  loadBalancerSourceRanges:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- else }}
+  type: {{ .Values.service.type }}
+  {{- end }}
+  {{- with .Values.service.externalIPs }}
+  externalIPs:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```
[the remark this answers is not in the document]
Comment (d3adb5) [reply application/templates/service.yaml]: Resolved in e90b5d3.

File: application/templates/service.yaml [hunk 2]
```diff
@@ -7,20 +7,35 @@ metadata:
...
+  type: ClusterIP
+  {{- with .Values.service.clusterIP }}
   clusterIP: {{ . }}
-{{- end }}
+  {{- end }}
+  {{- else if eq .Values.service.type "LoadBalancer" }}
+  type: {{ .Values.service.type }}
+  {{- with .Values.service.loadBalancerIP }}
+  loadBalancerIP: {{ . }}
+  {{- end }}
```
[the remark this answers is not in the document]
Comment (d3adb5) [reply application/templates/service.yaml:23]: Added as a feature in 5488bbc.

File: application/templates/service.yaml [hunk 3]
```diff
@@ -7,20 +7,35 @@ metadata:
...
-{{ toYaml .Values.service.additionalLabels | indent 4 }}
-{{- end }}
-{{- if .Values.service.annotations }}
+  {{- if .Values.service.additionalLabels }}
+  {{ toYaml .Values.service.additionalLabels | indent 4 }}
+  {{- end }}
+  {{- if .Values.service.annotations }}
   annotations:
 {{ include "application.tplvalues.render" ( dict "value" .Values.service.annotations  "context" $ ) | indent 4 }}
 {{- end }}
```
[the remark this answers is not in the document]
Comment (d3adb5) [reply application/templates/service.yaml]: Done in 0bd9dab.

**Comment by d3adb5** [comment]
@rasheedamir @aslafy-z I pushed commits addressing the issues I raised in my last review. Since it would be presumptuous of me to give myself a green light, could either of you take a quick look and merge if possible?

**Review by aslafy-z — APPROVED** [review]
LGTM, thank you @d3adb5!

**Pull request merged**